.github/bin/run-clang-tidy.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+set -u
+
+TMPDIR="${TMPDIR:-/tmp}"
+
+readonly PARALLEL_COUNT=$(nproc)
+readonly FILES_PER_INVOCATION=5
+
+readonly FILES_TO_PROCESS=${TMPDIR}/clang-tidy-files.list
+readonly TIDY_OUT=${TMPDIR}/clang-tidy.out
+
+# Using clang-tidy-11 as it is the last one still checking for
+# google-runtime-references, non-const references in parameters - which is
+# IMO better than the relaxed Google styleguide rules later.
+readonly CLANG_TIDY=clang-tidy-11
+hash ${CLANG_TIDY} || exit 2  # make sure it is installed.
+
+echo ::group::Build compilation database
+
+# TODO(hzeller): this might need modification if Surelog installed elsewhere.
+cmake -B build .
+
+find src -name "*.h" -o -name "*.cc" > ${FILES_TO_PROCESS}
+
+echo "::endgroup::"
+
+echo "::group::Run ${CLANG_TIDY} on $(wc -l < ${FILES_TO_PROCESS}) files"
+
+cat ${FILES_TO_PROCESS} \
+    | xargs -P${PARALLEL_COUNT} -n ${FILES_PER_INVOCATION} -- \
+            ${CLANG_TIDY} --quiet 2>/dev/null \
+            > ${TIDY_OUT}.tmp
+
+# Only modify at the end, so we can inspect the file manually while a new run
+# of this script is in progress.
+mv ${TIDY_OUT}.tmp ${TIDY_OUT}
+
+cat ${TIDY_OUT}
+
+echo "::endgroup::"
+
+if [ -s ${TIDY_OUT} ]; then
+
+    # Tidy results were non-empty. Put a summary into a separate group.
+    echo "::group::Summary"
+    # Give a nice overview of how many counts of which problem found.
+    sed 's|\(.*\)\(\[[a-zA-Z.-]*\]$\)|\2|p;d' < ${TIDY_OUT} | sort | uniq -c | sort -n
+    echo "::endgroup::"
+
+    echo "::error::There were clang-tidy warnings. Please fix"
+    exit 1
+fi
+
+echo "No clang-tidy complaints.😎"
```

Comment:
Emoji's in a TUI application repo?

Reply:
I do like that terminals these days are pretty good in doing all kinds of unicode characters.

Reply:
I've found it to still be inconsistent, e.g. opening stuff inside tmux vs outside has weird results. The ChromeOS SSH application has a lot of quirks with these types of characters too.
How about just a traditional :-) style?

Reply:
Done.

(For living inside the terminal, you might appreciate one of my projects https://timg.sh/ ... albeit also making use of special Unicode characters )

Reply:
Haha, wow, that's impressive. Has science gone too far?